client/extensions/woocommerce/state/sites/settings/email/actions.js

```diff
@@ -103,14 +103,21 @@ export const emailSettingsSubmitSettings = ( siteId, settings ) => dispatch => {
 
 	dispatch( settingsSubmit( siteId ) );
 
+	// disable if user has emptied the input field
```

Comment:
It seems like some of the code in actions here (actually making the requests, or data changing like this) might make more sense in the data-layer. Maybe not necessarily in this PR, but it might be worth doing a bit of refactoring to make sure it's consistent with other new code we are introducing.

Reply:
Sure! I know little about how data layer is structured but I am more than happy to learn. Could you point me to a place in code where data-layer is used?

Reply:
Sure! A good example is the orders code: https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/data-layer/orders/index.js 

An action is dispatched like normal: https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/sites/orders/actions.js#L29

A handler is registered for that action: https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/data-layer/orders/index.js#L35

The fetch: https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/data-layer/orders/index.js#L46

Data coming back: https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/data-layer/orders/index.js#L54

---

This P2 post (p4TIVU-6oC-p2) was also useful to me when first learning about the data-layer.

Reply:
OK I will tackle the refactor in a separate branch. Thanks for the tip and the links.